aeron-archive/src/main/java/io/aeron/archive/client/AeronArchive.java

```diff
@@ -1576,6 +1581,38 @@ public long getRecordedLength(final long recordingId)
         }
     }
 
+    /**
+     * Get the id of the Archive.
+     *
+     * @return the id of the Archive.
+     */
+    public long archiveId()
```

Comment:
Should this be cached in `AeronArchive`?

Reply:
It is cached. See below.

Reply:
Missed that.  A style note, I personally like the early return approach for this sort of thing, e.g:
```java
    if (Aeron.NULL_VALUE != archiveId)
    {
        return achiveId;
    }
        
    ensureOpen();
    ensureNotReentrant();

    lastCorrelationId = aeron.nextCorrelationId();

    if (!archiveProxy.archiveId(lastCorrelationId, controlSessionId))
    {
        throw new ArchiveException("failed to send get recorded length request");
    }

    archiveId = pollForResponse(lastCorrelationId);
    return archiveId;
```
Not required change for merging though.